docs/en/observability/profiling-self-managed-ops.asciidoc

```diff
@@ -185,44 +186,80 @@ When scaling out the number of replicas, depending on the load balancer that you
 This ensures that the load is evenly distributed across all replicas.
 
 [discrete]
-[[profiling-upgrade-backend-bin]]
-== Upgrade the backend binaries
+[[profiling-self-managed-upgrade]]
+== Upgrades
 
-Upgrade the backend binaries whenever you upgrade the rest of the Elastic stack.
-While we try to keep backward compatibility between two consecutive minor version, we may introduce changes to the data format that require the applications to be at the same version of Elasticsearch and Kibana.
+Upgrading a self-hosted stack involves upgrading the backend applications and the agent.
+We recommend to upgrade the backend first, followed by the agent. In this scenario, if the backend has problems,
+rolling back to the previous version is the safest option and does not require the agent to be downgraded.
+
+NOTE: It is usually best to have the same version of the agent and the backend deployed together.
+
+We strive to maintain backward compatibility between minor versions.
+We may introduce changes to the data format that require the agent and backend to be at the same version.
+When a breaking change in the protocol is introduced, the profiling agents that are not up to date will stop sending data.
+The agent logs will report an error message indicating that the backend is not compatible with the agent (or vice versa).
 
 The upgrade process steps vary depending on the installation method used.
+You may have a mixture of installation methods, e.g. the backend is deployed on ECE and the agents are deployed on Kubernetes.
+In that case, refer to the specific sections (backend/agent) in each method.
+
+IMPORTANT: Depending on your infrastructure setup, upgrading the backend may also update the endpoint exposed by the collector.
+In that case, amend the configuration of the agent accordingly, to have it connect to the new endpoint upon upgrade.
```

Comment:
```suggestion
Make sure to amend the agent configuration to connect to the new endpoint upon upgrade.
```

Reply:
A thing here: most of the times the endpoint does not change.
Should we be more explicit that only if the endpoint changes, the agent config needs to be amended?

Reply:
Sure, we can leave the "In this case," and say "In this case, amend the agent configuration to connect to the new endpoint upon upgrade."